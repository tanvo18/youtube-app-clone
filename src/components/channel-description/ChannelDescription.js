import React, { Component } from 'react';
import ChannelDescriptionStyled from './ChannelDescriptionStyled';
import PropTypes from 'prop-types';

class ChannelDescription extends Component {
  render() {
    return (
      <ChannelDescriptionStyled>
        <p>{this.props.channelTitle}</p>
        <p>{this.props.publishedAt}</p>
        <p>{this.props.description}</p>
      </ChannelDescriptionStyled>
    );
  }
}

ChannelDescription.defaultProps = {
  channelTitle: '',
  publishedAt: '',
  description: ''
};

ChannelDescription.propTypes = {
  channelTitle: PropTypes.string,
  publishedAt: PropTypes.string,
  description: PropTypes.string
};

export default ChannelDescription;
