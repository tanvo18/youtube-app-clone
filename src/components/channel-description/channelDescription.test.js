import React from 'react';
import ChannelDescription from './ChannelDescription';
import renderer from 'react-test-renderer';

const props = {
  channelTitle: 'Funny channel',
  publishedAt: 'Oct 15th 18',
  description: 'Download / stream BEAUTIFUL: https://Bazzi.lnk.to/BeautifulxCamilaAY \nCOSMIC: https://Bazzi.lnk.to/COSMICAY\n\nDirected by Jason Koenig https://www.instagram.com/jkoephoto/ \n\nSubscribe for more official content from Bazzi: https://Bazzi.lnk.to/Subscribe\n\nFollow Bazzi\nhttps://facebook.com/BazziWorldwide\nhttps://instagram.com/bazzi\nhttps://twitter.com/bazzi\nhttps://soundcloud.com/bazziworldwide\nhttp://bazziofficial.com\n\nThe official YouTube channel of Atlantic Records artist Bazzi. Subscribe for the latest music videos, performances, and more.'
};

describe('test channelDescription', () => {
  const wrapper = shallow(<ChannelDescription {...props} />);

  it('test render channelTitle', () => {
    expect(wrapper.find('p').at(0).text()).toBe('Funny channel');
    expect(wrapper.find('p').at(0).text()).not.toBe('Funny ');
  });

  it('test render publishedAt', () => {
    expect(wrapper.find('p').at(1).text()).toBe('Oct 15th 18');
  });

  it('test not pass props', () => {
    const wrapper = shallow(<ChannelDescription />);
    expect(wrapper.find('p').at(0).text()).toBe('');
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<ChannelDescription {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});