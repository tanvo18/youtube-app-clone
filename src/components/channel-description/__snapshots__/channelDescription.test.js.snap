// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test channelDescription should render correctly 1`] = `
<div
  className="sc-bdVaJa hooZEu"
>
  <p>
    Funny channel
  </p>
  <p>
    Oct 15th 18
  </p>
  <p>
    Download / stream BEAUTIFUL: https://Bazzi.lnk.to/BeautifulxCamilaAY 
    COSMIC: https://Bazzi.lnk.to/COSMICAY
    
    Directed by Jason Koenig https://www.instagram.com/jkoephoto/ 
    
    Subscribe for more official content from Bazzi: https://Bazzi.lnk.to/Subscribe
    
    Follow Bazzi
    https://facebook.com/BazziWorldwide
    https://instagram.com/bazzi
    https://twitter.com/bazzi
    https://soundcloud.com/bazziworldwide
    http://bazziofficial.com
    
    The official YouTube channel of Atlantic Records artist Bazzi. Subscribe for the latest music videos, performances, and more.
  </p>
</div>
`;
